google/self_link_helpers.go

```diff
@@ -28,6 +28,17 @@ func compareSelfLinkRelativePaths(k, old, new string, d *schema.ResourceData) bo
 	return false
 }
 
+// This method should only be used if the self link references a global resource.
+func compareSelfLinkResourceNames(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
The `network` field can be a `self_link` or the name of the network.
This method works when the field can be either `name` or `self_link`.

If we were using `compareSelfLinkRelativePaths`, then we would get a diff if we specify the network using its name in the config.

`mynetwork != project/global/networks/mynetwork`

I updated the comment to make it clearer.